src/UI/class-settings-page.php

```diff
@@ -757,6 +739,38 @@ public function print_binary_radio_tag( array $args ): void {
 		<?php
 	}
 
+	/**
+	 * Prints out multiple selection in the form of checkboxes
+	 *
+	 * @param array $args The arguments for the checkboxes.
+	 * @return void
+	 */
+	public function print_multiple_checkboxes( array $args ): void {
+		$options        = $this->parsely->get_options();
+		$select_options = $args['select_options'];
+		$id             = esc_attr( $args['option_key'] );
+		$name           = Parsely::OPTIONS_KEY . "[$id]";
+
+		if ( isset( $args['help_text'] ) ) {
+			echo '<div class="parsely-form-controls" data-has-help-text="true">';
+		}
+		if ( isset( $args['requires_recrawl'] ) && true === $args['requires_recrawl'] ) {
+			echo '<div class="parsely-form-controls" data-requires-recrawl="true">';
+		}
+
+		foreach ( $select_options as $key => $val ) {
+			$selected = in_array( $val, $options[ $args['option_key'] ], true );
+			echo sprintf( "<p><input type='checkbox' name='%s[]' id='%s[]' value='%s' ", esc_attr( $name ), esc_attr( $name ), esc_attr( $key ) );
+			echo checked( true === $selected, true, false );
+			echo sprintf( " /> <label for='%s_%s'>%s</label></p>", esc_attr( $id ), esc_attr( $key ), esc_html( $val ) );
```

Comment:
I am a bit confused with this:
  * on line 763: `id='%s[]'` => `esc_attr( $name )`
  * here: the `for` attribute does not match the `input`'s ID or am I getting this wrong?

Reply:
You got it right, I've now changed the `for` attribute.